src/TransitionViz.tsx

```diff
@@ -98,12 +98,17 @@ export const TransitionViz: React.FC<{
     return null;
   }
 
+  // extra check if the transition might be blocked by the 'in' property...
+  const isBlocked = !!definition.in && !state.matches(definition.in);
```

Comment:
How this is handled in XState's core suggests that this is not that simple:
https://github.com/statelyai/xstate/blob/89fcc6d47f1890f09752d73d086fdd158898c194/packages/core/src/StateNode.ts#L852-L863

Reply:
I think you are right. My code only works if state path is given, and not if it is a state id (#...). I am working on a generic solution right now.

Reply:
I added special handling for 'custom state ids'. I tested it with the (adapted) machine defined in #328 .